core/src/main/scala/org/apache/spark/sql/rapids/tool/AppBase.scala

```diff
@@ -426,15 +426,19 @@ abstract class AppBase(
         // Get ReadSchema of each Node and sanitize it for comparison
         val trimmedNode = trimSchema(ReadParser.parseReadNode(node).schema)
         readSchema.contains(trimmedNode)
-      }).filter(ReadParser.isScanNode(_)).head
+      }).filter(ReadParser.isScanNode(_))
 
-      dataSourceInfo += DataSourceCase(sqlID,
-        scanNode.id,
-        meta.getOrElse("Format", "unknown"),
-        meta.getOrElse("Location", "unknown"),
-        meta.getOrElse("PushedFilters", "unknown"),
-        readSchema
-      )
+      // If the ReadSchema is empty or if the Scan is not supported, then we don't need to
+      // add it to the dataSourceInfo
```

Comment:
In this case, what does not supported mean?  It still could be useful to log data source information even if it isn't running on GPU so we have a complete picture of data sources being used by the application.

Reply:
This is a special case if the eventlogs have  Photon reads.  ScanNode is `PhotonScan` and we currently check if ScanNode starts with either `Scan` or `NativeScan`.  Since we don't qualify any Photon operators yet, I updated it as above. Other unsupported datasource information will be captured since they will have `Scan` in ScanNode. I will update the above comment to include PhotonScan.

Reply:
We would see PhotonScan in unsupportedOperators.csv output file.